packages/glimmer-runtime/lib/modifier/interfaces.ts

```diff
@@ -4,15 +4,16 @@ import { DynamicScope } from '../environment';
 import { Destroyable } from 'glimmer-util';
 
 export interface ModifierManager<T> {
+  create(args: EvaluatedArgs, dynamicScope: DynamicScope, dom: DOMChanges): T;
   // At initial render, the modifier gets a chance to install itself on the
   // element it is managing. It can also return a bucket of state that
   // it could use at update time. From the perspective of Glimmer, this
   // is an opaque token.
-  install(element: Element, args: EvaluatedArgs, dom: DOMChanges, dynamicScope: DynamicScope): T;
+  install(element: Element);
```

Comment:
Need to pass the `T` here, also I _think_ we want to pass `dom: DOMChanges` here instead of the constructor (we hypothesized some scenarios where you might get a different thing each time, e.g. RehydrationDOMChanges -> DOMChanges)